### PR TITLE
lua-language-server: move to by-name; clean derivation

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

- **lua-language-server: move to by-name**
- **lua-language-server: clean derivation**

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
@ofborg build lua-language-server

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355032`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
I don't think the community builder can use ditto, either.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355032`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

File: pkgs/by-name/lu/lua-language-server/package.nix [hunk 1]
```diff
@@ -28,20 +27,20 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    CoreFoundation
-    Foundation
-    ditto
+    darwin.ditto
```
Comment (khaneliman) [inline pkgs/by-name/lu/lua-language-server/package.nix:30]:
```suggestion
    # aligned_alloc
    (darwinMinVersionHook "10.15")
```

File: pkgs/by-name/lu/lua-language-server/package.nix [hunk 2]
```diff
@@ -4,9 +4,8 @@
   fetchFromGitHub,
   ninja,
   makeWrapper,
-  CoreFoundation,
-  Foundation,
-  ditto,
+  darwin,
```
Comment (khaneliman) [inline pkgs/by-name/lu/lua-language-server/package.nix:7]:
```suggestion
  darwinMinVersionHook,
```

**Review by khaneliman — COMMENTED** [review]
We could add to this:
https://github.com/NixOS/nixpkgs/blob/23200937042bc534e15277654ae02da8c8c41fb5/pkgs/by-name/lu/lua-language-server/package.nix#L53-L56

```bash
      sed -i compile/ninja/macos.ninja \
        -e '/c++/s,$cc,clang++,' \
        -e '/test.lua/s,= .*,= true,' \
        -e '/ldl/s,$cc,clang++,' \
        -e '/ditto/s,ditto,cp -r,'
```

or we could add `rsync` as a dependency if we really need the extra metadata? Not too familiar with ditto.

File: pkgs/by-name/lu/lua-language-server/package.nix [hunk 3]
```diff
@@ -4,9 +4,8 @@
   fetchFromGitHub,
   ninja,
   makeWrapper,
-  CoreFoundation,
-  Foundation,
-  ditto,
+  darwinMinVersionHook,
```
Comment (khaneliman) [inline pkgs/by-name/lu/lua-language-server/package.nix:7]:
```suggestion
  apple-sdk_11
  darwinMinVersionHook,
```

File: pkgs/by-name/lu/lua-language-server/package.nix [hunk 4]
```diff
@@ -28,20 +27,21 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    CoreFoundation
-    Foundation
-    ditto
+    # aligned_alloc
+    (darwinMinVersionHook "10.15").ditto
```
Comment (khaneliman) [inline pkgs/by-name/lu/lua-language-server/package.nix:31]:
```suggestion
    # aligned_alloc
    apple-sdk_11
    (darwinMinVersionHook "10.15")
```

**Review by khaneliman — COMMENTED** [review]
Add these to the darwin patch

```bash
      substituteInPlace compile/ninja/macos.ninja \
        --replace-fail "ditto" "rsync -a"

      substituteInPlace scripts/writer.lua \
        --replace-fail "ditto" "rsync -a"
```

File: pkgs/by-name/lu/lua-language-server/package.nix (hunk 4 above)
Comment (GaetanLepage) [reply pkgs/by-name/lu/lua-language-server/package.nix]: Done !

**Comment by khaneliman** [comment]
> Add these to the darwin patch
> 
> ```shell
>       substituteInPlace compile/ninja/macos.ninja \
>         --replace-fail "ditto" "rsync -a"
> 
>       substituteInPlace scripts/writer.lua \
>         --replace-fail "ditto" "rsync -a"
> ```

This part is what will fix the ditto error

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355032`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355032`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

**Review by teto — APPROVED** [review]

File: pkgs/by-name/lu/lua-language-server/package.nix [hunk 5]
```diff
@@ -28,36 +34,49 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    CoreFoundation
-    Foundation
-    ditto
+    # aligned_alloc
```
Comment (SuperSandro2000) [inline pkgs/by-name/lu/lua-language-server/package.nix:37]: What's up with alligned_alloc?

Comment (khaneliman) [reply pkgs/by-name/lu/lua-language-server/package.nix:37]: Reason why we have to target a non default sdk

**Pull request merged**